misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/FormattingTreeModifier.java

```diff
@@ -4492,18 +4496,23 @@ private void preserveIndentation(boolean value) {
      * @param token token of which the indentation is required.
      */
     private int getPreservedIndentation(Token token) {
-        int position = token.lineRange().startLine().offset();
+        LinePosition startLinePos = token.lineRange().startLine();
+        int position = startLinePos.offset();
+        int startLine = startLinePos.line();
+        for (Token invalidToken : token.leadingInvalidTokens()) {
+            LinePosition invalidTokenStartLinePos = invalidToken.lineRange().startLine();
+            if (invalidTokenStartLinePos.line() == startLine) {
```

Comment:
According to the codcov, we need to add a test for `invalidTokenStartLinePos.line() != startLine`

Reply:
The line shown in the codecov is L4508 which is a close brace token. We do reach here for the test. I think codecov warning was mistaken.

Reply:
@poorna2152 can u confirm it as well.

Reply:
Added a test